tools/extract_baserom.c

```diff
@@ -0,0 +1,1614 @@
+    "shrine_scene",
+    "shrine_room_0",
+    "turibori_scene",
+    "turibori_room_0",
+    "shrine_n_scene",
+    "shrine_n_room_0",
+    "shrine_r_scene",
+    "shrine_r_room_0",
+    "hakaana2_scene",
+    "hakaana2_room_0",
+    "gerudoway_scene",
+    "gerudoway_room_0",
+    "gerudoway_room_1",
+    "gerudoway_room_2",
+    "gerudoway_room_3",
+    "gerudoway_room_4",
+    "gerudoway_room_5",
+    "hairal_niwa_n_scene",
+    "hairal_niwa_n_room_0",
+    "bowling_scene",
+    "bowling_room_0",
+    "hakaana_ouke_scene",
+    "hakaana_ouke_room_0",
+    "hakaana_ouke_room_1",
+    "hakaana_ouke_room_2",
+    "hylia_labo_scene",
+    "hylia_labo_room_0",
+    "souko_scene",
+    "souko_room_0",
+    "souko_room_1",
+    "souko_room_2",
+    "miharigoya_scene",
+    "miharigoya_room_0",
+    "mahouya_scene",
+    "mahouya_room_0",
+    "takaraya_scene",
+    "takaraya_room_0",
+    "takaraya_room_1",
+    "takaraya_room_2",
+    "takaraya_room_3",
+    "takaraya_room_4",
+    "takaraya_room_5",
+    "takaraya_room_6",
+    "ganon_sonogo_scene",
+    "ganon_sonogo_room_0",
+    "ganon_sonogo_room_1",
+    "ganon_sonogo_room_2",
+    "ganon_sonogo_room_3",
+    "ganon_sonogo_room_4",
+    "ganon_demo_scene",
+    "ganon_demo_room_0",
+    "besitu_scene",
+    "besitu_room_0",
+    "face_shop_scene",
+    "face_shop_room_0",
+    "kinsuta_scene",
+    "kinsuta_room_0",
+    "ganontikasonogo_scene",
+    "ganontikasonogo_room_0",
+    "ganontikasonogo_room_1",
+    "test01_scene",
+    "test01_room_0",
+    "bump_texture_static",
+    "anime_model_1_static",
+    "anime_model_2_static",
+    "anime_model_3_static",
+    "anime_model_4_static",
+    "anime_model_5_static",
+    "anime_model_6_static",
+    "anime_texture_1_static",
+    "anime_texture_2_static",
+    "anime_texture_3_static",
+    "anime_texture_4_static",
+    "anime_texture_5_static",
+    "anime_texture_6_static",
+    "softsprite_matrix_static",
+};
+
+int main(void)
+{
+
+    FILE *rom = fopen("baserom.z64", "rb");
+    FILE *outFile;
+    unsigned int virtualStart;
+    unsigned int virtualEnd;
+
+
+
+    if (rom == NULL)
+    {
+        fprintf(stderr, "Could not open baserom.z64\n");
+        exit(1);
+    }
+
+    for (unsigned int i = 0; i < DMA_TABLE_ENTRIES; i++)
+    {
+        char outFileName[45] = "baserom/";
+        fseek(rom, DMA_TABLE_OFFSET + (DMA_ENTRY_SIZE * i), SEEK_SET);
+
+        if (fread(&virtualStart, sizeof(unsigned int), 1, rom) != 1)
+        {
+            fprintf(stderr, "Could not read start address for file %s\n", sDmaMgrFileNames[i]);
+            exit(1);
+        }
+
+        virtualStart = __bswap_32(virtualStart);
+
+        if (fread(&virtualEnd, sizeof(unsigned int), 1, rom) != 1)
+        {
+            fprintf(stderr, "Could not read end address for file %s\n", sDmaMgrFileNames[i]);
+            exit(1);
+        }
+
+        virtualEnd = __bswap_32(virtualEnd);
+
+        void *data = malloc(virtualEnd - virtualStart);
```

Comment:
You are leaking lots of RAM here. That _may_ be a problem to the guys using WSL, since they always complain about WSL eating all their RAM, but I don't know

Reply:
I had a `free(data)` on a the original but I guess it got lost when moving it into the repo.

Reply:
Thinking about it, doing a `malloc` in each iteration of the loop may be slow. You could just do a huge `malloc` before the for loop and re-use that  pointer over and over again. Just do a big enough `enough` malloc to ensure every file will always fit in it.